plugins/chat/app/services/services.md

```diff
@@ -0,0 +1,153 @@
+### Services
+
+There are many different types of services which need to do many different
+things. We have a lot of them already in chat, a selection:
+
+* [ChatChannelArchiveService](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/lib/chat_channel_archive_service.rb)
+* [ChatTranscriptService](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/lib/chat_transcript_service.rb)
+* [ChatMessageCreator](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/lib/chat_message_creator.rb)
+* [ChatMessageUpdater](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/lib/chat_message_updater.rb)
+* [ChatMessageDestroyer](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/app/services/chat_message_destroyer.rb)
+* [ChatMessagePublisher](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/app/services/chat_publisher.rb)
+* [MessageMover](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/lib/message_mover.rb)
+* [SlackCompatibility](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/lib/slack_compatibility.rb)
+* [ChatChannelFetcher](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/lib/chat_channel_fetcher.rb)
+* [ChatMessageReactor](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/lib/chat_message_reactor.rb)
+* [ChatChannelMembershipManager](https://github.com/discourse/discourse/blob/f34b4f67b1f15c51ecba9ad73b4e39515cf9e988/plugins/chat/lib/chat_channel_membership_manager.rb)
+
+These services all do different things. Some are purely for data fetching (e.g.
+ChatChannelFetcher), others are used for MessageBus publishing (e.g.
+ChatMessagePublisher), while still more handle complicated processes that
+include doing things like this (taking ChatMessageCreator as an example):
+
+* Validating input parameters
+* Raising errors based on failed validation, access control, and other issues
+* Querying the database
+* Validating current database record state
+* Changing database record state
+* Triggering DiscourseEvent events
+* Calling other services
+* Enqueueing jobs
+* Checking permissions with Guardian
+* Logging, both to STDOUT and to things like StaffActionLogger
+* Interacting with MessageBus
+
+When writing services, we also need to think about how we handle failure and
+success states, especially for the Controllers that are most often calling them
+(though of course services can and do call other services). With that in mind we
+need to handle each different function of services differently to make it easier
+for the developer.
```

Comment:
I think there are two other places where it's going to be called a lot too:
- specs
- jobs